src/vos/vos_ts.h

```diff
@@ -0,0 +1,629 @@
+/**
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/**
+ * Record timestamp table
+ * vos/vos_ts.h
+ *
+ * Author: Jeff Olivier <jeffrey.v.olivier@intel.com>
+ */
+
+#ifndef __VOS_TS__
+#define __VOS_TS__
+
+#include <vos_tls.h>
+
+struct vos_ts_info {
+	/** Least recently accessed index */
+	uint32_t		ti_lru;
+	/** Most recently accessed index */
+	uint32_t		ti_mru;
+	/** Type identifier */
+	uint32_t		ti_type;
+	/** mask for hash of negative entries */
+	uint32_t		ti_cache_mask;
+	/** Number of entries in cache for type (for testing) */
+	uint32_t		ti_count;
+};
+
+struct vos_ts_entry {
+	struct vos_ts_info	*te_info;
+	/** Uniquely identifies the record */
+	void			*te_record_ptr;
+	/** Uniquely identifies the parent record */
+	uint32_t		*te_parent_ptr;
+	/** negative entry cache */
+	uint32_t		*te_miss_idx;
+	/** Low read time */
+	daos_epoch_t		 te_ts_rl;
+	/** High read time */
+	daos_epoch_t		 te_ts_rh;
```

Comment:
Could you explain a bit what's "Low read time" & "High read time"?

Reply:
Is that any clearer.   I think Li Wei will be updating the VOS readme for MVCC with more information on how they are supposed to be used.

Reply:
Yes, and I've read the updated README, thanks a lot. I think it was the term of 'high' & 'low' confused me.